jvm/src/test/scala/net/teralytics/terahex/Generators.scala

```diff
@@ -75,4 +76,25 @@ object Generators {
     level <- levels
     cells <- listOfN(level, subCells)
   } yield root.copy(cells = cells)
+
+  case class HexagonIdWithLevel(hexagonId: BigInt, level: Int)
+
+  def bigInt(numDigits: Int): Gen[BigInt] =
+    (0 until numDigits).foldLeft(const(BigInt(0))) {
+      (gen, pos) =>
+        for {
+          digit <- choose(0, 9)
+          curr <- gen
+        } yield curr + digit * (Math.pow(10, pos).toLong)
```

Comment:
For this you could generate random characters from '0' to '9' and then use the String-argument BigInt constructor.